File: python/day03/p1.py
from string import ascii_letters

if __name__ == "__main__":
    result = 0
    with open("input.txt", "r") as f:
        for line in f.read().split("\n"):
            s1 = set(line[: len(line) // 2])
            s2 = set(line[len(line) // 2 :])

            for c in s1.intersection(s2):
                result += ascii_letters.index(c) + 1
        print(result)

File: rust/day03/p1.rs
use std::{collections::HashSet, fs};

fn main() {
    let lines = fs::read_to_string("input.txt").unwrap();

    let x: u32 = lines
        .trim()
        .split("\n")
        .map(|rucksack| {
            let (r1, r2) = rucksack.split_at(rucksack.len() / 2);
            let hsr1: HashSet<char> = r1.chars().collect();
            let hsr2: HashSet<char> = r2.chars().collect();
            let r: u32 = hsr1
                .intersection(&hsr2)
                .map(|c| {
                    let mut ord = *c as u32;
                    if ord > 96 {
                        ord -= 96
                    } else {
                        ord -= 38
                    }
                    ord
                })
                .sum();

            r
        })
        .sum();

    println!("{}", x)
}
